From 870b0368d615714c105176ef03098ce4d5dae5e2 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Wed, 15 Jul 2015 09:54:41 -0700
Subject: [PATCH 002/131] pci: Add is_pcierc element to struct pci_bus
Git-commit: c8683dc58b4e7b89e99f8abb6c022e946929f33b
Patch-mainline: Queued in subsystem maintainer repository
Git-repo: git://git.kernel.org/pub/scm/linux/kernel/git/rric/linux.git thunder/master-v4.1

... and use is to force only_one_child() to return true.

Needed because the ThunderX PCIe RC cannot be identified by existing methods.

Signed-off-by: David Daney <david.daney@cavium.com>

Signed-off-by: Matthias Brugger <mbrugger@suse.com>

---
 drivers/pci/probe.c |    2 ++
 include/linux/pci.h |    1 +
 2 files changed, 3 insertions(+)

--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -1804,6 +1804,8 @@ static int only_one_child(struct pci_bus
 {
 	struct pci_dev *parent = bus->self;
 
+	if (bus->is_pcierc)
+		return 1;
 	if (!parent || !pci_is_pcie(parent))
 		return 0;
 	if (pci_pcie_type(parent) == PCI_EXP_TYPE_ROOT_PORT)
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -478,6 +478,7 @@ struct pci_bus {
 	struct bin_attribute	*legacy_io; /* legacy I/O for this bus */
 	struct bin_attribute	*legacy_mem; /* legacy mem */
 	unsigned int		is_added:1;
+	unsigned int		is_pcierc:1;
 };
 
 #define to_pci_bus(n)	container_of(n, struct pci_bus, dev)
